app/javascript/components/widget-controller.jsx

```diff
@@ -36,7 +38,7 @@ const getWidgets = gql`
 export const WidgetController = ({ client, cityName }) => {
   const [newWidgetId, setNewWidgetId] = useState(0);
 
-  const fetchPolicy = newWidgetId === 0 ? 'network-only' : 'cache-first';
+  const fetchPolicy = newWidgetId === 0 ? 'cache-first' : 'network-only';
```

Comment:
I'm not following this change. `newWidgetId` will be 0 on first load, the ternary will evaluate to true, and `network-only` will be used.

Reply:
I agree that but when I change it to cache first, the widgets did not show loading and work as expected. I ran into the fresh browser and checked it. Is there a suggestion on what I can look up to?

Reply:
This suggests that issue might be related to the apollo cache and requires further investigation to understand what the actual culprit is.

https://www.apollographql.com/docs/react/data/queries/#supported-fetch-policies